### PR TITLE
bluefox usb3 support

### LINKED ISSUE #121: Support for Bluefox USB3

### PR DESCRIPTION
**Pull request by g3force**
Resolve #121 

This PR adds support for USB3 Bluefox cameras from matrix-vision.
It is also a preparation for upcoming PRs: 
 * Require OpenCV > 2 for color conversion (this breaks compatibility with neural calibrator, which was never actively used and would need to be migrated to new OpenCV API first)
 * Switch from travis to circleci to build on Ubuntu 18.04 with more recent library versions
 * Allow to build without DC1394, because the Bluefox SDK has issues with some combinations of libraries (its still on by default)

### CODE REVIEW

File: src/app/capture_thread.cpp [hunk 1]
```diff
@@ -176,7 +194,7 @@ void CaptureThread::kill() {
 
 bool CaptureThread::init() {
   capture_mutex.lock();
-  bool res = capture->startCapture();
+  bool res = capture != nullptr && capture->startCapture();
```
Comment (joydeep-b) [inline src/app/capture_thread.cpp]: Need parens to make intent clear.
```c++
res = (capture != nullptr) && capture->startCapture();
```

File: src/app/capture_thread.cpp [hunk 2]
```diff
@@ -189,7 +207,7 @@ bool CaptureThread::init() {
 
 bool CaptureThread::stop() {
   capture_mutex.lock();
-  bool res = capture->stopCapture();
+  bool res = capture != nullptr && capture->stopCapture();
```
Comment (joydeep-b) [inline src/app/capture_thread.cpp]: Need parens.
```c++
bool res = (capture != nullptr) && capture->stopCapture();
```

File: src/app/capture_thread.cpp [hunk 3]
```diff
@@ -202,14 +220,16 @@ bool CaptureThread::stop() {
 
 bool CaptureThread::reset() {
   capture_mutex.lock();
-  bool res = capture->resetBus();
+  bool res = capture != nullptr && capture->resetBus();
```
Comment (joydeep-b) [inline src/app/capture_thread.cpp]: Need parens

File: src/shared/capture/capture_bluefox2.cpp [hunk 4]
```diff
@@ -27,8 +27,9 @@ CaptureBlueFox2::CaptureBlueFox2(VarList * _settings,int default_camera_id, QObj
 CaptureBlueFox2::CaptureBlueFox2(VarList * _settings,int default_camera_id) : CaptureInterface(_settings)
 #endif
 {
-  cam_id = default_camera_id;
+  cam_id = (unsigned int) default_camera_id;
```
Comment (joydeep-b) [inline src/shared/capture/capture_bluefox2.cpp]: static cast is preferable to make intent clear:
```c++
cam_id = static_cast<unsigned int>(default_camera_id);
```

File: src/shared/capture/capture_bluefox2.cpp [hunk 5]
```diff
@@ -327,11 +329,15 @@ bool CaptureBlueFox2::startCapture()
...
     mutex.lock();
   #endif
+
+  if(pDevMgr == nullptr) {
+    pDevMgr = new DeviceManager();
+  }
     
   //grab current parameters:
-  cam_id = v_cam_bus->getInt();
+  cam_id = (unsigned int) v_cam_bus->getInt();
```
Comment (joydeep-b) [inline src/shared/capture/capture_bluefox2.cpp]: static_cast

File: CMakeLists.txt [hunk 6]
```diff
@@ -1,5 +1,17 @@
 project(ssl-vision)
+set(OpenGL_GL_PREFERENCE LEGACY)
+set(USE_NEURAL_CALIBRATION FALSE CACHE BOOL "Enable neural network calibration (requires OpenCV 2, not compatible to later versions)")
+set(USE_DC1394 TRUE CACHE BOOL "Compile with DC1394 (firewire cameras)")
 set(USE_QT5 FALSE CACHE BOOL "Use Qt5 instead of Qt4")
+set(USE_mvIMPACT FALSE CACHE BOOL "Compile with mvImpact (bluefox cameras)")
+
+if(USE_DC1394 AND USE_mvIMPACT)
+	message(FATAL_ERROR "DC1394 and mvImpact are not compatible: mvImpact crashes when creating device manager")
```
[the remark this answers is not in the document]
Comment (g3force) [reply CMakeLists.txt]: I don't think it is that important, especially as we now decided to use the FLIR cameras. Also, I think it is ok to assume that Firewire und USB cameras are usually not run together.

**Comment by g3force** [comment]
I changed the CMakefile to check for the right OpenCV version and fixed a compile issue when having the pylon sdk installed.

**Pull request merged**